app/src/main/java/com/rosario/covid19/ui/HomeActivity.kt

```diff
@@ -86,9 +88,9 @@ class HomeActivity : AppCompatActivity(), View.OnClickListener {
                 calendar.set(Calendar.YEAR, year)
                 calendar.set(Calendar.MONTH, month)
                 calendar.set(Calendar.DAY_OF_MONTH, day)
-                tv_date.text = Util().userFormatDate(calendar)
+                dateTitleText.text = Util().userFormatDate(calendar)
                 selectedDate = Util().dateFormat(calendar)
-                homeViewModel.getReport(Util().dateFormat(calendar))
+                homeViewModel.getReport(selectedDate)
             },
             yearCal,
             monthCal,
```

Comment:
You can handle all the calendar logic into an extension function or the Utils class to keep the code cleaner